File: AllLongestStrings.js
function allLongestStrings(inputArray) {
    maxLen = 0;
    for (i = 0; i < inputArray.length; i++) {
        if (inputArray[i].length > maxLen) {
            maxLen = inputArray[i].length;
        }
    }
    
    longestStringsArray = [];
    for (i = 0; i < inputArray.length; i++) {
        if (inputArray[i].length == maxLen) {
            longestStringsArray.push(inputArray[i]);
        }
    }
    return longestStringsArray;
}

File: intro_algorithms/AllLongestStrings/allLongestStrings.py
def allLongestStrings(inputArray):
    strLength = 0
    for arrElem in inputArray:
        if len(arrElem) >= strLength:
            strLength = len(arrElem)
    
    arr = []
    for arrElem in inputArray:
        if len(arrElem) == strLength:
            arr.append(arrElem)
    return arr